ui/src/tool-panel/geometry-alignment/geometry-alignment-linking-candidates.tsx

```diff
@@ -147,6 +178,63 @@ export const GeometryAlignmentLinkingReferenceLineCandidates: React.FC<
         }
     }, [selectedLayoutReferenceLine]);
 
+    const referenceLineElements = referenceLines?.map((line) => {
+        const isSelected = line.id == selectedLayoutReferenceLine?.id;
+        const ref = referenceLineRefs.find((r) => r.id == line.id);
+        const trackNumber = trackNumbers?.find((tn) => tn.id === line.trackNumberId);
+
+        if (!ref || !trackNumber) {
+            return <React.Fragment key={line.id} />;
+        }
+
+        if (
+            referenceLineSearchInput &&
```

Comment:
Useamman iffin ja kommentoitujen lohkojen sijaan tekisin noista ehdoista muuttujat (tai funktiot jos tuntuu fiksummalta), jolloin nimet korvaa kommentin + yhdistäisin nämä skippaavat lohkot yhdeksi. En ole varma luinko kaikki booleanit oikein päin mutta siis jotain tyyliin:
```
const tnExists = ref && trackNumber
const tnMatchesSearch = !referenceLineSearchInput || trackNumber?.number?.toLoweCase()?.includes(referenceLineSearchInput)
const tnIsOfficialWithoutGeometry = referenceLineSearchInput && !line.foundWithBoundingBox && line.draftType === 'OFFICIAL'
if (!trackNumberExists || !tnMatchesSearch || tnIsOfficialWithoutGeometry) {
  return <React.Fragment key={line.id} />;
}
```

Reply:
+1 nimettyjen logiikkamuuttujien käyttämiselle, näitä voisi olla useammassakin paikassa tässä koodikannassa.

Tuo yhdistelty iffittely toimii toiston (koodikohinan) vähentämiseksi, joskin kognitiivinen kompleksisuus tulkita "tyhjä palautustilanne" hieman kasvaa. Yhdistelty iffi ohjaa myös jatkossa lisäämään "skippaa tämä"-tyyliset tarkistukset samaan paikkaan, kun taas sanoisin erillisten iffien ohjaamaan rajatumpien funktioiden muodostamiseen (ohjelmoija alkaisi miettimään varsinaista rakennetta, kun koodi selvästi pitenee lisätarkistusten johdosta). Tiedäppä sitten kumpi on parempi tapa 🤷, tässä yhteydessä tuo yhdistely ainakin toimii.

On myös tullut touhuttua sen verran runsaasti koodin kanssa, joka palauttaa kutsujalle/lokittaa mikä erikoistilanne tapahtui (jos "oikeaan arvoon" ei funktiossa päädytty tai tapahtui vaikkapa virhe), että tulee tuo erillisten iffien käyttäminen herkemmin meikäläiselle mieleen. Vaikka samaan "tyhjään palautusarvoon" päädyttäisiinkiin, niin debuggaaminen saattaa helpottua, kun tiettyihin haaroihin päädytään ainoastaan tietyssä tilanteessa eikä potentiaalisesti useammassa (riippuu työkaluista mikä debuggausmenetelmä on toimivin).

Reply:
Iffittelyjä vähennetty ja pyritty muuttamaan helpommin ymmärrettäviksi käyttämällä nimettyjä logiikkamuuttujia ja yhtä tyhjän komponentin palautuskohtaa.